src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -1564,11 +1573,85 @@ bool DataWriterImpl::deadline_timer_reschedule()
         return false;
     }
 
+    if (next_deadline_us <= std::chrono::steady_clock::now())
+    {
+        // Ignore uninitialized or stale deadlines; don't arm to avoid a spurious first miss
+        return false;
+    }
```

Comment:
TODO: double check before merging whether this is actually needed or the condition was given due to a flaw already solved.